include/particles/particle_bbar.h

```diff
@@ -0,0 +1,144 @@
+  ParticleBbar(Index id, const VectorDim& coord);
+
+  //! Construct a particle using B-bar method with id, coordinates and status
+  //! \param[in] id Particle id
+  //! \param[in] coord coordinates of the particle
+  //! \param[in] status Particle status (active / inactive)
+  ParticleBbar(Index id, const VectorDim& coord, bool status);
+
+  //! Destructor
+  ~ParticleBbar() override{};
+
+  //! Delete copy constructor
+  ParticleBbar(const ParticleBbar<Tdim>&) = delete;
+
+  //! Delete assignment operator
+  ParticleBbar& operator=(const ParticleBbar<Tdim>&) = delete;
+
+  //! Type of particle
+  std::string type() const override {
+    return (Tdim == 2) ? "P2DBBAR" : "P3DBBAR";
+  }
+
+ protected:
+  //! particle id
+  using ParticleBase<Tdim>::id_;
+  //! coordinates
+  using ParticleBase<Tdim>::coordinates_;
+  //! Reference coordinates (in a cell)
+  using ParticleBase<Tdim>::xi_;
+  //! Cell
+  using ParticleBase<Tdim>::cell_;
+  //! Cell id
+  using ParticleBase<Tdim>::cell_id_;
+  //! Nodes
+  using ParticleBase<Tdim>::nodes_;
+  //! Status
+  using ParticleBase<Tdim>::status_;
+  //! Material
+  using ParticleBase<Tdim>::material_;
+  //! Material id
+  using ParticleBase<Tdim>::material_id_;
+  //! State variables
+  using ParticleBase<Tdim>::state_variables_;
+  //! Neighbour particles
+  using ParticleBase<Tdim>::neighbours_;
+  //! Volumetric mass density (mass / volume)
+  using Particle<Tdim>::mass_density_;
+  //! Mass
+  using Particle<Tdim>::mass_;
+  //! Volume
+  using Particle<Tdim>::volume_;
+  //! Size of particle
+  using Particle<Tdim>::size_;
+  //! Size of particle in natural coordinates
+  using Particle<Tdim>::natural_size_;
+  //! Stresses
+  using Particle<Tdim>::stress_;
+  //! Strains
+  using Particle<Tdim>::strain_;
+  //! dvolumetric strain
+  using Particle<Tdim>::dvolumetric_strain_;
+  //! Volumetric strain at centroid
+  using Particle<Tdim>::volumetric_strain_centroid_;
+  //! Strain rate
+  using Particle<Tdim>::strain_rate_;
+  //! dstrains
+  using Particle<Tdim>::dstrain_;
+  //! Velocity
+  using Particle<Tdim>::velocity_;
+  //! Displacement
+  using Particle<Tdim>::displacement_;
+  //! Particle velocity constraints
+  using Particle<Tdim>::particle_velocity_constraints_;
+  //! Free surface
+  using Particle<Tdim>::free_surface_;
+  //! Free surface
+  using Particle<Tdim>::normal_;
+  //! Set traction
+  using Particle<Tdim>::set_traction_;
+  //! Surface Traction (given as a stress; force/area)
+  using Particle<Tdim>::traction_;
+  //! Shape functions
+  using Particle<Tdim>::shapefn_;
+  //! dN/dX
+  using Particle<Tdim>::dn_dx_;
+  //! dN/dX at cell centroid
+  using Particle<Tdim>::dn_dx_centroid_;
+  //! Map of scalar properties
+  using Particle<Tdim>::scalar_properties_;
+  //! Map of vector properties
+  using Particle<Tdim>::vector_properties_;
+  //! Map of tensor properties
+  using Particle<Tdim>::tensor_properties_;
+  //! Pack size
+  using Particle<Tdim>::pack_size_;
```

Comment:
Just call the variables that you want to use in this class

Reply:
Okay, I've just removed unnecessary ones.